components/raftstore/src/store/peer.rs

```diff
@@ -3987,12 +3976,36 @@ where
             }
         }
 
-        if self.raft_group.raft.has_pending_conf()
-            || self.raft_group.raft.pending_conf_index > index
-        {
+        // It's safe to transfer leader to a target peer that has already applied the
+        // configuration change, even if the current leader has not yet applied
+        // it. For more details, refer to the issue at:
+        // https://github.com/tikv/tikv/issues/17363#issuecomment-2404227253.
+        if self.raft_group.raft.pending_conf_index > index {
+            info!(
+                "not ready to transfer leader, target peer has an unapplied conf change";
+                "region_id" => self.region_id,
+                "target_peer_id" => peer_id,
+                "pending_conf_index" => self.raft_group.raft.pending_conf_index,
+                "leader_applied_index" => self.raft_group.raft.raft_log.applied,
+                "target_applied_index" => index
+            );
             return Some("pending conf change");
         }
 
+        if self.raft_group.raft.has_pending_conf() {
```

Comment:
Is this log necessary ? We've already allowed this condition in this pr, do we need to log it here ?

Reply:
This is a special case, and we may sometimes need to detect when it occurs. Without this log, is there another way to identify it?

Reply:
Got.